devito/symbolics/printer.py

```diff
@@ -31,6 +32,11 @@ def _print_Function(self, expr):
         # There exist no unknown Functions
         if expr.func.__name__ not in self.known_functions:
             self.known_functions[expr.func.__name__] = expr.func.__name__
+        if isinstance(expr, devito.types.SymbolicArray):
```

Comment:
this is odd...maybe SymbolicArray is just lacking proper ``__repr__`` and ``__str__`` which in turn will be used by the sympy printer

Reply:
`_print_Function` doesn't use `__str__` but manually puts stuff in parentheses